### ISSUE TITLE
[SCA] Add support for rules that test numerical ordering 

### ISSUE THREAD

**Issue by JuantAldea** [wazuh/wazuh#3374]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.9 | SCA | -- | -- | -- |

There are several rules in SCA the perform numerical comparison by rejecting/acceping patterns that match, for instance, a rule for matching a number that is less than 365 would resemble
```
- f:/etc/login.defs -> 
     r:^PASS_MAX_DAYS\s*\t*\d$|
       ^PASS_MAX_DAYS\s*\t*\d\d$|
       ^PASS_MAX_DAYS\s*\t*\d\d\d$
  && !r:^PASS_MAX_DAYS\s*\t*366|
        ^PASS_MAX_DAYS\s*\t*367|
        ^PASS_MAX_DAYS\s*\t*368|
        ^PASS_MAX_DAYS\s*\t*369|
        ^PASS_MAX_DAYS\s*\t*37\d|
        ^PASS_MAX_DAYS\s*\t*38\d|
        ^PASS_MAX_DAYS\s*\t*39\d|
        ^PASS_MAX_DAYS\s*\t*4\d\d|
        ^PASS_MAX_DAYS\s*\t*5\d\d|
        ^PASS_MAX_DAYS\s*\t*6\d\d|
        ^PASS_MAX_DAYS\s*\t*7\d\d|
        ^PASS_MAX_DAYS\s*\t*8\d\d|
        ^PASS_MAX_DAYS\s*\t*9\d\d
```
(but the whole thing in one line) which is confusing to say the least.

The aim is to support rules such as
`n:PASS_MAX_DAYS\s*\t*(\d+) compare <= 365`
that would take the captured regex and compare it as requested.

Supported operators are the usual ordering operators `==, !=, <, <=, >, >=`

**Comment by JuantAldea** [comment]
This has been implemented as part of #3286

**Comment by chemamartinez** [comment]
This issue has been added at (https://github.com/wazuh/wazuh/commit/7fc48ee1d34fcbd25f49ea746a6bf751842e398a) and reviewed. It is included as a part of the [SCA refactor](https://github.com/wazuh/wazuh/issues/3249) included in 3.10.

Closing it.
